server/cluster/cluster.go

```diff
@@ -124,7 +151,9 @@ func (c *Cluster) updateCacheLocked(
 		shardMetaList := shards[shardID]
 		var nodes []*clusterpb.Node
 		for _, shardMeta := range shardMetaList {
-			nodes = append(nodes, c.nodesCache[shardMeta.Node].meta)
+			if node := c.nodesCache[shardMeta.Node]; node != nil {
```

Comment:
It is confusing that this `node` is not the element of the `nodes` slice, and maybe we should rename the `nodes` to `nodeInfos` or `nodeMetas`.